src/lib/mina_base/account_update.ml

```diff
@@ -1137,14 +1137,23 @@ module Body = struct
 
     let deriver obj =
       let open Fields_derivers_zkapps in
+      let may_use_token =
+        needs_custom_js
+          ~js_type:
+            (js_record
+               [ ("parentsOwnToken", js_layout bool)
+               ; ("inheritFromParent", js_layout bool)
+               ] )
+          ~name:"MayUseToken" May_use_token.deriver
+      in
```

Comment:
nit: it might be a bit cleaner to add this logic to `May_use_token.deriver`, not here